### PR TITLE
FIX problem when search boolean (Yes/No) extrafields with value 0 (No)

### PR DESCRIPTION
**Pull request by hregis**

### CODE REVIEW

**Comment by hregis** [comment]
@eldy same in v15 to develop

File: htdocs/core/tpl/extrafields_list_search_input.tpl.php [hunk 1]
```diff
@@ -47,7 +47,7 @@
 					if (in_array($typeofextrafield, array('link', 'sellist', 'text', 'html'))) {
 						$morecss = 'maxwidth200';
 					}
-					echo $extrafields->showInputField($key, (empty($search_array_options[$search_options_pattern.$tmpkey]) ? '' : $search_array_options[$search_options_pattern.$tmpkey]), '', '', $search_options_pattern, $morecss, 0, $extrafieldsobjectkey, 1);
+					echo $extrafields->showInputField($key, ((empty($search_array_options[$search_options_pattern.$tmpkey]) && $search_array_options[$search_options_pattern.$tmpkey] != 0) ? '' : $search_array_options[$search_options_pattern.$tmpkey]), '', '', $search_options_pattern, $morecss, 0, $extrafieldsobjectkey, 1);
```
Comment (eldy) [inline htdocs/core/tpl/extrafields_list_search_input.tpl.php:50]: According to line 47, we are not in the case of bolean valu here but in the case of html, text, link or sellist. 
Did i miss something ?

Comment (hregis) [reply htdocs/core/tpl/extrafields_list_search_input.tpl.php:50]: @eldy boolean use this showInputField !

Comment (hregis) [reply htdocs/core/tpl/extrafields_list_search_input.tpl.php:50]: @eldy create a boolean for propal, go to list and search "No", the value "No" is not kept, with my correction the value "No" remains following the search

**Comment by hregis** [comment]
@eldy ping

**Review by eldy — COMMENTED** [review]
Seems fixed in recent version, but in a different way. To avoid pb in merging, i will close this pr. If we need a backport, using the same fix than the one done for recent version must be used.

**Pull request closed without merging**